ssl/ssl_sess.c

```diff
@@ -163,8 +165,14 @@ SSL_SESSION *ssl_session_dup(SSL_SESSION *src, int ticket)
     if (dest->lock == NULL)
         goto err;
 
-    if (src->peer != NULL)
-        X509_up_ref(src->peer);
+    if (!CRYPTO_new_ex_data(CRYPTO_EX_INDEX_SSL_SESSION, dest, &dest->ex_data))
```

Comment:
Yes, or it could be moved up a few lines?
Anyway if the new callback gets called the object pointer will be
in a half-initialized state.